site/app/libraries/FileUtils.php

```diff
@@ -174,6 +174,24 @@ public static function emptyDir($dir) {
         }
     }
 
+    /**
+     * checks to see if directory is empty, if so return true
+     *
+     * @param string $dir
+     */
+    public static function isEmptyDir($dir) {
+        if ($handle = opendir($dir)) {
+            while (false !== ($file = readdir($handle))) {
+                if ($file != '.' && $file != '..') {
+                    closedir($handle);
+                    return false;
+                }
+            }
+            closedir($handle);
+        }
+        return true;
+    }
```

Comment:
Would it not be cleaner to use `return count(scandir(...)) > 2` for this?

Reply:
I saw that scandir did look at all of the files, perhaps I am wrong. But that is why I chose to write a function to check without looking at all of the files if the directory is empty.

Reply:
Viane is correct. scandir would defeat the point of this PR, which is to make it so we _don't_ have to look at every file to determine if the course_materials directory is empty.